apps/frontend/src/pages/quiz/libs/components/quiz-form/quiz-form.tsx

```diff
@@ -94,39 +125,35 @@ const QuizForm: React.FC<Properties> = ({ onNext }: Properties) => {
 						</div>
 					) : (
 						category?.map((question) => {
-							const answerOptions = question.answers.map(
-								({ label, value }) => ({
-									label,
-									value: value.toString(),
-								}),
-							);
+							const answerOptions = question.answers.map(({ id, label }) => ({
+								label,
+								value: id.toString(),
+							}));
 
 							return (
 								<QuizQuestion
 									control={control}
 									key={question.id}
 									label={question.label}
-									name={question.id.toString()}
+									name={`answer${question.id.toString()}`}
 									options={answerOptions}
 								/>
 							);
 						})
 					)}
 				</div>
 				<div className={styles["form-footer"]}>
+					{currentCategoryIndex !== ZERO && (
+						<div className={styles["button-container"]}>
+							<Button
+								label="BACK"
+								onClick={handlePreviousStep}
+								variant="secondary"
+							/>
+						</div>
+					)}
 					<div className={styles["button-container"]}>
-						<Button
-							label="BACK"
-							onClick={handlePreviousStep}
-							variant="secondary"
-						/>
-					</div>
-					<div className={styles["button-container"]}>
-						{isLast ? (
-							<Button label="CONTINUE" onClick={onNext} type="submit" />
-						) : (
-							<Button label="NEXT" onClick={handleNextStep} type="submit" />
-						)}
+						<Button label="NEXT" type="submit" variant="primary" />
```

Comment:
I see that user can't go the next page not answering current questions, but you should also add isDisabled property to Button.

Reply:
Submit button still doesn't have isDisabled property.